src/promptflow/promptflow/contracts/flow.py

```diff
@@ -539,6 +539,12 @@ def deserialize(data: dict) -> "NodeVariants":
         return NodeVariants(default_variant_id=data.get("default_variant_id", ""), variants=variants)
 
 
+class FlowLanguage(str, Enum):
+    """The enum of tool source type."""
+
+    Python = "python"
```

Comment:
If we use an enum, it would take more effort to add new type, and hit backward compatibility problem this is why our first version uses str but not enum

Reply:
we can't do backward compatibility for language support right? We will need a new proxy for each new language. Whatever, I will revert it to a string and transfer this enum class to constants to refer.